jota/src/main/java/org/iota/jota/model/Transaction.java

```diff
@@ -499,59 +409,167 @@ public String toTrytes() {
                 + Converter.trytes(attachmentTimestampLowerBoundTrits)
                 + Converter.trytes(attachmentTimestampUpperBoundTrits)
                 + this.getNonce();
-        return trytes;
     }
 
-    /**
-     * Initializes a new instance of the Signature class.
-     * 
-     * @param trytes
-     */
-    public void transactionObject(final String trytes) {
+    public static class Builder {
+        private transient ICurl customCurl;
+        private String hash;
+        private String signatureFragments;
+        private String address;
+        private long value;
+        private String obsoleteTag;
+        private long timestamp;
+        private long currentIndex;
+        private long lastIndex;
+        private String bundle;
+        private String trunkTransaction;
+        private String branchTransaction;
+        private String nonce;
+        private Boolean persistence;
+        private long attachmentTimestamp;
+        private String tag;
+        private long attachmentTimestampLowerBound;
+        private long attachmentTimestampUpperBound;
+
+        /**
+         * Start builder with curl mode CURL_P81 as default
+         */
+        public Builder() {
+            this.customCurl = SpongeFactory.create(SpongeFactory.Mode.CURL_P81);
+        }
+
+        public Builder curl(ICurl curl) {
+            this.customCurl = curl;
+            return this;
+        }
+
+        public Builder hash(String hash) {
+            this.hash = hash;
+            return this;
+        }
+
+        public Builder signatureFragments(String signatureFragments) {
+            this.signatureFragments = signatureFragments;
+            return this;
+        }
+
+        public Builder address(String address) {
+            this.address = address;
+            return this;
+        }
+
+        public Builder value(long value) {
```

Comment:
Codacy found an issue: [Public method and constructor comments are required](https://app.codacy.com/manual/Iota-java/iota-java/pullRequest?prid=6319461)

Reply:
This is documented, why isn't it outdated?